django/core/handlers/base.py

```diff
@@ -121,7 +121,13 @@ def _get_response(self, request):
         if response is None:
             wrapped_callback = self.make_view_atomic(callback)
             try:
-                response = wrapped_callback(request, *callback_args, **callback_kwargs)
+                response = wrapped_callback(
+                    request, *callback_args, **callback_kwargs)
+            except TypeError:
```

Comment:
What if something else throws this `TypeError`? If we want to give a message like this at all, I think we need to find a better way of ensuring the case you mentioned is causing it.